src/ObjCRuntime/Protocol.cs

```diff
@@ -24,6 +24,18 @@ public Protocol (string name)
 				throw new ArgumentException (String.Format ("'{0}' is an unknown protocol", name));
 		}
 
+		public Protocol (Type type)
+		{
+			this.handle = Class.GetHandle (type);
```

Comment:
This is not correct.

1. First check that the type is an interface, otherwise it can't be a protocol:

        if (!type.IsInterface)
            throw new ArgumentException (string.Format ("'{0}' is not a protocol.", type.FullName));

2. Then get the Protocol attribute, and get the name of the protocol from there:

        var protocols = type.GetCustomAttributes (typeof (ProtocolAttribute), false);
        if (protocols.Count == 0)
            throw ...

        var protocol = protocols [0];
        var protocolName = protocol.Name;
        handle = objc_getProtocol (protocolName);
        if (handle == IntPtr.Zero)
            throw ...